ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx

```diff
@@ -1,44 +1,42 @@
-import React from 'react';
-import { useHistory, useParams } from 'react-router-dom';
-import { Button, PageSection } from '@patternfly/react-core';
+import React, { useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import { Divider, PageSection, Title } from '@patternfly/react-core';
 
-import { complianceEnhancedCoveragePath } from 'routePaths';
+import useRestQuery from 'hooks/useRestQuery';
+import { getComplianceProfileResults } from 'services/ComplianceResultsService';
 
+import PageTitle from 'Components/PageTitle';
 import CoveragesToggleGroup from './CoveragesToggleGroup';
 import CoveragesPageHeader from './CoveragesPageHeader';
+import ProfileChecksTable from './ProfileChecksTable';
 
 function ProfileChecksPage() {
-    const history = useHistory();
     const { profileName } = useParams();
 
+    const fetchProfileChecks = useCallback(
+        () => getComplianceProfileResults(profileName),
+        [profileName]
+    );
+    const { data: profileChecks, loading: isLoading, error } = useRestQuery(fetchProfileChecks);
+
     return (
         <>
+            <PageTitle title="Compliance coverage - Profile checks" />
             <CoveragesPageHeader />
             <PageSection>
                 <CoveragesToggleGroup tableView="checks" />
             </PageSection>
-            <PageSection variant="light">
-                <div>ProfileChecksPage</div>
-                <Button
-                    onClick={() => {
-                        history.push(
-                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/clusters`
-                        );
-                    }}
-                    variant="primary"
-                >
-                    Go to all clusters (ProfileClustersPage)
-                </Button>
-                <Button
-                    onClick={() => {
-                        history.push(
-                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/checks/test`
-                        );
-                    }}
-                    variant="primary"
-                >
-                    Go to single check (CheckDetailsPage)
-                </Button>
+            <PageSection variant="default">
+                <PageSection variant="light">
+                    <Title headingLevel="h2">Profile results</Title>
+                    <Divider />
+                    <ProfileChecksTable
+                        isLoading={isLoading}
+                        error={error}
+                        profileChecks={profileChecks?.profileResults ?? []}
```

Comment:
Questions for my understanding, not necessarily change requests.
* Given use of table state and `TbodyUnified` element within `ProfileChecksTable` component, is `?? []` optional chaining redundant here? That is, functionally. I see it is needed because type the prop.
* Maybe more to the root, what are pro and con of table state within table component or in page component?

Here is a recent example of the latter:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/NodeCvePage.tsx#L68-L73
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/AffectedNodesTable.tsx#L69-L74

Reply:
I'm not sure I understand your first question, are you asking why do optional chaining? or why have the nullish coalescing operator (`??`)? `profileChecks` could be undefined here so the optional chaining is required and I want to ensure `profileChecks` inside the table is an array so I don't have to do my `undefined` checks there. The prop name could be confusing, maybe `profileChecksResults` would be more accurate?

As far as table state, I think there are pro and cons to both. My thoughts were more centralized data management at the page level and reduced complexity at the table level. I can see an argument being made for data fetching to live in the table component though. The examples you links mimic my pattern, no? Passed as a single prop but data fetching still happens at the parent level

Reply:
Yeah, I meant nullish coalescing.

Reply:
Regardless of the question, `profileChecksResults` is a nice improvement on the name.

Maybe I was double unclear. Thursday afternoon pretending to be Friday afternoon again this week?

What if page is responsible both for fetching and for getting corresponding table state for the table?

The alternative came to mind because of paradox between what might be and what is:
* On one side of the coin, the component could use the prop directly, so better safe than sorry.
* On other side of the coin, the component uses the prop indirectly via `data` in `renderer` function.

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Components/TableStateTemplates/TbodyUnified.tsx#L65-L67

Although `data: undefined | DataType[]` it is an array in `'POLLING'` and `'COMPLETE'` states:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/getTableUIState.ts#L11-L19

As an aside, `Array.isArray(data)` might be more expressive than `data` here:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/getTableUIState.ts#L59-L69

Reply:
> Maybe I was double unclear. Thursday afternoon pretending to be Friday afternoon again this week?

Don't worry, I can relate..

I need to take a bit of time to really understand how `getTableUIState` actually works under the hood. I think I see where you're coming from now though, essentially should table state live in the same place that the data does. I think that makes sense. I'm going to keep this pattern here for now but try the alternative in `ProfileClustersPage` to get a better feel. Although moving `getTableUIState` up is sounding better in my head